lib/clients/authenticated.js

```diff
@@ -163,21 +185,14 @@ class AuthenticatedClient extends PublicClient {
             callback(undefined, response, totalDeletedOrders);
           }
           return totalDeletedOrders;
-        })
-        .catch(err => {
-          if (callback) {
-            callback(err);
-          }
-          throw err;
         });
     }.call(this);
 
     if (callback) {
-      p.catch(() => {});
```

Comment:
These conditions ensure that a promise is returned to the user if and only if they did not provide a callback, enforcing the mutually exclusive use of the callback API or promise API per method call. If the user is not dealing with promises, we need to add a blank `catch` handler on the (now internal-only) promise so that any rejected promise will not cause an `UnhandledPromiseRejectionWarning`.

While I like the aesthetics of `p.catch(callback)` much better, it runs into a potential problem with inversion of control; if the user's `callback` `throw`s, then it will either fail silently (on node < 6.6.0), cause an `UnhandledPromiseRejectionWarning` (currently, which is difficult to meaningfully trace during debugging), or terminate the node process (in future node).